pool.go

```diff
@@ -147,7 +185,8 @@ func (p *Pool) Get(ctx context.Context) (*ClientConn, error) {
 	}
 
 	wrapper := ClientConn{
-		pool: p,
+		pool:        p,
+		concurrency: &ConcurrencyCounter{},
 	}
 	select {
 	case wrapper = <-clients:
```

Comment:
Am I reading this right - that you create a `wrapper` var right above this, and then assign to it? If so, is the wrapper var above just... unused? Can't be.

Reply:
This is from the original implementation of the pool; it may not be necessary but modifying it doesn't seem necessary in this PR